fix 'no more updates from usbmuxd after the last device disconnects' problem

sam80180 opened the pull request:
> Whether this PR is eventually merged or not, Sonic will thank you very much for your contribution. 
> 
> 无论此PR最终是否合并，Sonic组织都非常感谢您的贡献。

### Checklist

- [x] The title starts with fix, fea, or doc. | 标题为fix、feat或doc开头。
- [x] I have checked that there are no duplicate pull requests with this request. | 我已检查没有与此请求重复的拉取请求。
- [x] I have considered and confirmed that this submission is valuable to others. | 我已经考虑过，并确认这份呈件对其他人很有价值。
- [x] I accept that this submission may not be used. | 我接受此提交可能不会被使用。

### Description

When the last device disconnects, usbmuxd stops itself.
As a result, the 'sib devices listen' command (and any other function that uses usbmux listen function) no longer receives device status update.

Here's my workaround to this problem. Hope it helps.

sam80180 committed:
fix: fix 'no more responses from usbmuxd after the last device disconnects' problem

ZhouYixun commented:
In many places in sonic-agent, according to the output of fmt in sib, are you sure that changing to logrus will not affect it?

sam80180 commented:
`logrus` output is directed to `stderr`
(https://github.com/SonicCloudOrg/sonic-ios-bridge/blob/main/src/util/log.go#L14)

if you're worried, feel free to modify it to `fmt`

ZhouYixun commented:
> `logrus` output is directed to `stderr` (https://github.com/SonicCloudOrg/sonic-ios-bridge/blob/main/src/util/log.go#L14)
> 
> if you're worried, feel free to modify it to `fmt`

It will make some sonic-agent bugs, could you modify it to `fmt`? I think logrus should record our running status as a local file at the same time, but console should output fmt only, please check other files

sam80180 commented:
Could you be more specific where it might cause problems?

ZhouYixun commented:
> Could you be more specific where it might cause problems?

Some places rely on the console output of sib to do relevant logic. Of course, this processing method is not good enough. If it is changed to logrus, it will take more time for the console output to be output, which may cause some Agent problems.

sam80180 commented:
```
# save this as 'l.go'
package main
import (
	"fmt"
	"log"
	"github.com/SonicCloudOrg/sonic-ios-bridge/src/util"
	"github.com/sirupsen/logrus"
)
func main() {
	util.InitLogger("debug")
	log.Println("aaa")
	logrus.Info("bbb")
	fmt.Println("ccc")
}
```

then run `go run l.go 2>/dev/null`, you can see that `fmt` is not affected by `logrus`. therefore it should not affect agent's behavior.

sam80180 committed:
fix: fix standard log module output bug

sam80180 committed:
fix: use constant backoff algorithm

If all devices are disconnected for a long time, sib will wait forever if exponential backoff algorithm is used.

sam80180 committed:
fix: log with context data

sam80180 committed:
Merge branch 'SonicCloudOrg:main' into main

sam80180 committed:
fix: can choose to stop listening on Ctrl+C

sam80180 committed:
Merge branch 'SonicCloudOrg:main' into main

sam80180 committed:
fix: close usbmux client when not used

sam80180 commented:
> # save this as 'l.go'
> package main
> import (
> 	"fmt"
> 	"log"
> 	"github.com/SonicCloudOrg/sonic-ios-bridge/src/util"
> 	"github.com/sirupsen/logrus"
> )
> func main() {
> 	util.InitLogger("debug")
> 	log.Println("aaa")
> 	logrus.Info("bbb")
> 	fmt.Println("ccc")
> }

2024/07/01 10:46:10 aaa
ccc

sam80180 commented:
> 
> could you show console output? I am busy now, sorry

you should only see  
```
2024/07/01 10:46:10 aaa
ccc
```

ZhouYixun commented:
> > could you show console output? I am busy now, sorry
> 
> you should only see
> 
> ```
> 2024/07/01 10:46:10 aaa
> ccc
> ```

Thank you so much, have you test it with sonic-agent?

ZhouYixun commented:
Thank you very much for your help. I will test it again in the near future.
